.github/workflows/build-scan.yml

```diff
@@ -0,0 +1,19 @@
+name: build-scan
+on: 
+  # Since branch protection prevents pushing to development, pushes only happen when merging PRs, so this runs only on merges (there's no way to set GitHub Actions to run on merges)
+  push:
+    branches:
+      - development
+
+jobs:
+  build-scan:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: redhat-actions/buildah-build@v2
```

Comment:
What is built depends on where it's going. If a local artifact is we want because we intend to scan it locally (to GH), then this might be an option. Otherwise, we're looking at building/pushing to a container registry.

Reply:
All the container scan actions I've looked at so far involve building the container in one step, and then pushing to a registry is the second step. What I have here is the first step of the push-to-quay.io action: https://github.com/marketplace/actions/push-to-registry

Reply:
Is `quay.io` is the container scanning solution we're going with?

Reply:
There's no reason to limit this project to a single container scanning solution. See [OWASP Docker Security Cheat Sheet RULE #9 - Use static analysis tools](https://cheatsheetseries.owasp.org/cheatsheets/Docker_Security_Cheat_Sheet.html#rule-9-use-static-analysis-tools) for options to augment [the OCIO's Aqua Service](https://developer.gov.bc.ca/Platform-Services-Security/BC-Government-Aqua-Cloud-Service-Definition).

Reply:
I think I'm missing something. If we [build with `buildah` we can push with `podman`](https://developers.redhat.com/blog/2019/02/21/podman-and-buildah-for-docker-users#podman_and_container_images) and the output is still an [OCI Image](https://github.com/opencontainers/image-spec). @bdolor how are you imagining having a local copy of the immutable docker layers will be affected by this line and/or how might building locally affect our ability to push layers and tags to a container registry, be that Github, Quay, OCP4, etc.?

Reply:
@wenzowski No reason to assume a limitation has been made, so I gather what's missing for you is the context of this work, which is related to container scanning.

Reply:
@wenzowski the problem to solve is what container scanning solution we're using. The question about local artifacts is trying to discover what the intention was here. Without knowing it was drawing inspiration from a quay.io playbook, it was unclear what this was trying to do.

Reply:
> Is quay.io is the container scanning solution we're going with?

I understood the context of the work but not the question about Quay–per my comment I wanted to note that a decision to go with quay doesn't preclude use of the OCIO's Aqua Service.

Per rule #9 link above Quay's free engine is called Clair and Acqua's in called Trivy. There are community contributed GH actions for both. Like with git commits, docker layers are immutable objects as validated by checksum.

> What is built depends on where it's going. If a local artifact is we want because we intend to scan it locally (to GH), then this might be an option. Otherwise, we're looking at building/pushing to a container registry.

I understand the gap in clarity and why there would be difficulty understanding intention of the PR due to missing provenance of the approach. My point however was that choice of a scanning solution doesn't depend on where it's going since OCI images can be push/pulled from one registry to another and all–even when using the OpenShift Container Platform’s integrated registry, or an external registry like quay.io–all locations of an image can be used. Per rule #9 the best practice for the build is to be deterministic, so even if we re-build to support a particular scanning technology. However, the best practice would be to [scan images as both part of the CI pipeline, and again as part of the k8s lifecycle](https://sysdig.com/blog/image-scanning-admission-controller/). My understanding of the OCIO Aqua  work is enabling this best practice, thus the comment implying no need to choose between Quay/Clair a Aqua/Trivy. I would advise, however, that in order to best enable a shift left we would want to ensure the deploy-time scan is also run at integration-time, along with any additional integration-time checks and/or technologies enabled.

Reply:
So...if we're introducing build scanning by way of gh actions, I would posit that a "shift left" approach would be to scan twice–once locally so we don't push a bad image, and once when pulling the image so we don't pull down an image that has been compromised since it was scanned on its way into registry, or so we have a guardrail in case of CI breach and first scan isn't trustworthy